feature-libs/asm/customer-360/components/customer-360/customer-360.component.ts

```diff
@@ -0,0 +1,315 @@
+  @HostBinding('attr.role') role = 'dialog';
+  @HostBinding('attr.aria-modal') modal = true;
+  @HostBinding('attr.aria-labelledby') labelledby = 'asm-customer-360-title';
+  @HostBinding('attr.aria-describedby') describedby = 'asm-customer-360-desc';
+  @ViewChildren('tabHeaderItem') tabHeaderItems: QueryList<
+    ElementRef<HTMLElement>
+  >;
+  readonly cartIcon = ICON_TYPE.CART;
+  readonly closeIcon = ICON_TYPE.CLOSE;
+  readonly orderIcon = ICON_TYPE.ORDER;
+  readonly ticketIcon = ICON_TYPE.FILE;
+
+  globalMessageType = GlobalMessageType;
+
+  focusConfig: FocusConfig = {
+    trap: true,
+    block: true,
+    autofocus: '.cx-tab-header.active',
+    focusOnEscape: true,
+  };
+
+  tabs: Array<Customer360TabConfig>;
+
+  activeTab = 0;
+
+  currentTab: Customer360TabConfig;
+
+  customer: User;
+
+  customer360Tabs$: Observable<Array<Customer360Data | undefined>>;
+
+  customerOverview$: Observable<CustomerOverview | undefined>;
+
+  protected subscription = new Subscription();
+
+  protected showErrorAlert$ = new BehaviorSubject<boolean>(false);
+
+  protected showErrorTab$ = new BehaviorSubject<boolean>(false);
+
+  constructor(
+    protected customer360Config: Customer360Config,
+    protected customer360Facade: Customer360Facade,
+    protected launchDialogService: LaunchDialogService,
+    protected csAgentAuthService: CsAgentAuthService,
+    protected directionService: DirectionService
+  ) {
+    this.customerOverview$ = this.customer360Facade
+      .get360Data([
+        {
+          requestData: { type: Customer360Type.OVERVIEW },
+        },
+      ])
+      .pipe(
+        map((response) => {
+          const overviewItem = response?.value?.find(
+            (item) => item.type === Customer360Type.OVERVIEW
+          ) as Customer360Overview | undefined;
+          return overviewItem?.overview || undefined;
+        }),
+        catchError(() => {
+          this.showErrorAlert$.next(true);
+          return of(undefined);
+        })
+      );
+
+    this.tabs = customer360Config?.customer360?.tabs ?? [];
+    this.currentTab = this.tabs[0];
+  }
+
+  ngOnInit(): void {
+    this.subscription.add(
+      this.csAgentAuthService
+        .isCustomerSupportAgentLoggedIn()
+        .pipe(distinctUntilChanged())
+        .subscribe((loggedIn) => {
+          if (!loggedIn) {
+            this.launchDialogService.closeDialog('logout');
+          }
+        })
+    );
+
+    this.subscription.add(
+      this.launchDialogService.data$.subscribe((data) => {
+        this.customer = data.customer;
+      })
+    );
+    this.setTabData();
+  }
+
+  ngAfterViewInit(): void {
+    this.updateTabIndex(this.activeTab);
+  }
+
+  ngOnDestroy(): void {
+    this.subscription.unsubscribe();
+  }
+
+  get errorAlert$(): Observable<boolean> {
+    return this.showErrorAlert$.asObservable();
+  }
+
+  get errorTab$(): Observable<boolean> {
+    return this.showErrorTab$.asObservable();
+  }
```

Comment:
[MINOR REMARK] can be fixed in a patch release
Creating fresh observables in functions that are called in html template is suboptimal. In other words, let's not wrap it inside `get` function, but simply let's have a property:

```ts
errorAlert$ = this.showErrorAlert$.asObservable();
...
```

Why it was suboptimal (in terms of performance): on each change detection cycle of this component, the HTML template is evaluated and the functions of the getters used in the HTML tempalte are called.
Then a fresh observable is created (becasue calling $something.asObservable() - which creates a fresh observable). And then `async` pipe in template unsubscribes from the previous observable and subscribes to the new observable returned from this getter.

ADDITIONAL ACTION NEEDED:
Please review this PR for other similar places of creating a fresh observable within a function that is called in HTML template.